packages/amplify-cli-logger/src/AmplifyLogger.ts

```diff
@@ -0,0 +1,87 @@
+import winston, { Logger, format } from 'winston';
+import winstonDailyRotateFile from 'winston-daily-rotate-file';
+import { constants } from './constants';
+import { IAmplifyLogger } from './IAmplifyLogger';
+import { JSONUtilities } from 'amplify-cli-core';
+import { getLogFilePath, getLocalLogFilePath, getLogAuditFilePath, getLocalAuditLogFile } from './getLogFilePath';
+import { LocalProjectData, LogPayload, LogErrorPayload } from './Types';
+import { Redactor } from './Redactor';
+
+export class AmplifyLogger implements IAmplifyLogger {
+  logger: Logger;
+  format: winston.Logform.Format;
+  localProjectData!: LocalProjectData;
+  disabledAmplifyLogging: boolean = !!process.env.AMPLIFY_CLI_DISABLE_LOGGING;
+  options: any = {
+    mode: 0o640,
+  };
+  constructor() {
+    this.logger = winston.createLogger();
+    this.format = format.combine(format.timestamp(), format.splat(), format.printf(this.formatter));
+    if (!this.disabledAmplifyLogging) {
+      this.logger.add(
+        new winstonDailyRotateFile({
+          auditFile: getLogAuditFilePath(),
+          filename: getLogFilePath(),
+          datePattern: constants.DATE_PATTERN,
+          maxFiles: `${constants.MAX_FILE_DAYS}d`,
```

Comment:
A couple nits on the maintainability of this config object:
1. I try to not mix functions and constants. If you can export constants from your `getLogFilePath` above then it's better to stay consistent that way future refactors are done in one place.
2. Having a string literal along with a `d` formatter at the end is a problem with maintenance as well, especially when you've got this code lower on line 35. I can see a bug in the future where you change it in one place but not the other.

Reply:
I am generating the folder name in the getLogFilePath. Depending on if it's `amplify-dev` or `amplify`